invenio_oauthclient/models.py

```diff
@@ -152,9 +156,9 @@ def token(self):
     def update_token(self, token, secret):
         """Update token with new values."""
         if self.access_token != token or self.secret != secret:
-            self.access_token = token
-            self.secret = secret
-            db.session.commit()
+            with db.session.begin_nested():
+                self.access_token = token
+                self.secret = secret
```

Comment:
add `db.session.add(self)` in case somebody uses the function directly after initializing the object